streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamTaskTest.java

```diff
@@ -202,6 +206,8 @@ public void testProcessOrder() throws Exception {
         assertEquals(0, task.numBuffered());
         assertEquals(3, source1.numReceived);
         assertEquals(3, source2.numReceived);
+
+        cleanRun = true;
```

Comment:
I don't get why we care about cleanRun everywhere. Seems it is only used in the `cleanup` and why does it matter? How is it going to impact the test if we just pass `true` or `false` all the time? The way it is done seems very odd and adds noise